client/opentracing/src/main/java/io/opentracing/APMSpanBuilder.java

```diff
@@ -34,20 +34,24 @@
 
     private TraceRecorder recorder;
     private ContextSampler sampler;
+    private String serviceName;
+    private String buildStamp;
 
     /**
      * @param operationName The operation name
      * @param recorder The trace recorder
      */
-    APMSpanBuilder(String operationName, TraceRecorder recorder, ContextSampler sampler) {
+    APMSpanBuilder(String operationName, TraceRecorder recorder, ContextSampler sampler, String serviceName, String buildStamp) {
```

Comment:
agree - or by storing the relevant tags in a single map as mentioned above. Depends whether we want it coupled with the trace recorder - it feels separate concerns to me.